### PR TITLE
[EASY] add eth token

### PR DESCRIPTION
**Pull request by GoodDataBadGuy**
# Thank you for contributing to Spellbook!

Thank you for taking the time to submit code in Spellbook. A few things to consider:

- If you are a first-time contributor, please sign the CLA by copy & pasting **_exactly_** what the bot mentions in PR comment
- Refer to [docs](#spellbook-contribution-docs) section below to answer questions
- Dune team will review submitted PRs as soon as possible

### To speed up your development process in PRs, keep these tips in mind:

- Each commit to your feature branch will rerun CI tests ([see example](https://github.com/duneanalytics/spellbook/actions/runs/8202519819/job/22433451880?pr=5519))
    - This includes *all* modified models on your branch
    - This includes *all* history of the data
- Two tips for faster development iteration:
    - Ensure dbt is installed locally (refer to main `readme`) and run `dbt compile`
        - This will output raw SQL in `target/` directory to copy/paste and run on Dune directly for initial query testing
    - Hardcode a `WHERE` filter for only ~7 days of history on large source tables, i.e. `ethereum.transactions`
        - This will speed up the CI tests and output results quicker -- whether that's an error or fully successful run
        - Once comfortable with small timeframe, remove filter and let full history run

### Incremental model setup
- Make sure your unique key columns are *exactly* the same in the model config block, schema yml file, and seed match columns (where applicable)
- There cannot be nulls in the unique key columns
    - Be sure to double check key columns are correct or `COALESCE()` as needed on key column(s), otherwise the tests may fail on duplicates

### 🪄 Use the built CI tables for testing 🪄

Once CI completes, you can query the CI tables and errors in dune when it finishes running.
- For example:
    - In the `run initial models` and `test initial models`, there will be a schema that looks like this: `test_schema.git_dunesql_4da8bae_sudoswap_v2_base_pool_creations`
    - This can be temporarily queried in Dune for ~24 hours

Leverage these tables to perform QA testing on Dune query editor -- or even full test dashboards!

## Spellbook contribution docs

The [docs](docs) directory has been implemented to answer as many questions as possible. Please take the time to reference each `.md` file within this directory to understand how to efficiently contribute & why the repo is designed as it is 🪄

Example questions to be answered:

- What does each property in the [model config block](/docs/models/model_config_block.md) mean?
- What is the [CI test](/docs/ci_test/ci_test_overview.md) attached to PRs and how can I best utilize it?
- What are the Spellbook [best practices](/docs/general/best_practices.md)?

Please navigate through the [docs](/docs) directory to find as much info as you can.

**Note**: happy to take PRs to improve the docs, let us know 🤝

### CODE REVIEW

File: tokens/models/tokens/ethereum/tokens_ethereum_erc20.sql [hunk 1]
```diff
@@ -3089,4 +3089,8 @@ FROM (VALUES
     , (0x72831eebef4e3f3697a6b216e3713958210ae8cd, 'BLOB', 18)
     , (0xe4129c7b229812212f88d1bd6a223c45622e6b85, 'LOOT', 9)
     , (0x725b9787fc9daf3bfd3fa21ec1bd88315181d164, 'Expectation vs Reality', 18)
+    , (0xf3768D6e78E65FC64b8F12ffc824452130BD5394, 'KEROSENE', 18)
```
Comment (GoodDataBadGuy) [inline tokens/models/tokens/ethereum/tokens_ethereum_erc20.sql:3092]: https://etherscan.io/address/0xf3768D6e78E65FC64b8F12ffc824452130BD5394

File: tokens/models/tokens/ethereum/tokens_ethereum_erc20.sql [hunk 2]
```diff
@@ -3089,4 +3089,8 @@ FROM (VALUES
     , (0x72831eebef4e3f3697a6b216e3713958210ae8cd, 'BLOB', 18)
     , (0xe4129c7b229812212f88d1bd6a223c45622e6b85, 'LOOT', 9)
     , (0x725b9787fc9daf3bfd3fa21ec1bd88315181d164, 'Expectation vs Reality', 18)
+    , (0xf3768D6e78E65FC64b8F12ffc824452130BD5394, 'KEROSENE', 18)
+    , (0x82a605D6D9114F4Ad6D5Ee461027477EeED31E34, 'SNSY', 18)
```
Comment (GoodDataBadGuy) [inline tokens/models/tokens/ethereum/tokens_ethereum_erc20.sql:3093]: https://etherscan.io/address/0x82a605D6D9114F4Ad6D5Ee461027477EeED31E34

File: tokens/models/tokens/ethereum/tokens_ethereum_erc20.sql [hunk 3]
```diff
@@ -3089,4 +3089,8 @@ FROM (VALUES
     , (0x72831eebef4e3f3697a6b216e3713958210ae8cd, 'BLOB', 18)
     , (0xe4129c7b229812212f88d1bd6a223c45622e6b85, 'LOOT', 9)
     , (0x725b9787fc9daf3bfd3fa21ec1bd88315181d164, 'Expectation vs Reality', 18)
+    , (0xf3768D6e78E65FC64b8F12ffc824452130BD5394, 'KEROSENE', 18)
+    , (0x82a605D6D9114F4Ad6D5Ee461027477EeED31E34, 'SNSY', 18)
+    , (0x1e07ebA252499371c147d2A0F6aaecd6427dF41E, 'EMRLDP', 18)
```
Comment (GoodDataBadGuy) [inline tokens/models/tokens/ethereum/tokens_ethereum_erc20.sql:3094]: https://etherscan.io/address/0x1e07ebA252499371c147d2A0F6aaecd6427dF41E

File: tokens/models/tokens/ethereum/tokens_ethereum_erc20.sql [hunk 4]
```diff
@@ -3089,4 +3089,8 @@ FROM (VALUES
     , (0x72831eebef4e3f3697a6b216e3713958210ae8cd, 'BLOB', 18)
     , (0xe4129c7b229812212f88d1bd6a223c45622e6b85, 'LOOT', 9)
     , (0x725b9787fc9daf3bfd3fa21ec1bd88315181d164, 'Expectation vs Reality', 18)
+    , (0xf3768D6e78E65FC64b8F12ffc824452130BD5394, 'KEROSENE', 18)
+    , (0x82a605D6D9114F4Ad6D5Ee461027477EeED31E34, 'SNSY', 18)
+    , (0x1e07ebA252499371c147d2A0F6aaecd6427dF41E, 'EMRLDP', 18)
+    , (0x396c95abE154b3b2eD204Cf45c8726AA7ad47A4D, 'GDT', 18)
```
Comment (GoodDataBadGuy) [inline tokens/models/tokens/ethereum/tokens_ethereum_erc20.sql:3095]: https://etherscan.io/address/0x396c95abE154b3b2eD204Cf45c8726AA7ad47A4D

**Comment by jeff-dude** [comment]
gm @GoodDataBadGuy -- thank you for looking to add. are these recently deployed tokens? i'm asking because our new automated source may pick them up for us. for instance, i checked just now, one has already been added:
```
select *
from tokens.erc20
where contract_address in (
0xf3768D6e78E65FC64b8F12ffc824452130BD5394
,0x82a605D6D9114F4Ad6D5Ee461027477EeED31E34
,0x1e07ebA252499371c147d2A0F6aaecd6427dF41E
,0x396c95abE154b3b2eD204Cf45c8726AA7ad47A4D
)
```

i wonder if we can let the automated source do it's thing? not 100% guaranteed they'll all come through, but thought i'd ask.

**Comment by GoodDataBadGuy** [comment]
yes they are newly deployed token and I always try to add new deployed for a dashboard that I had - I need the token name to make it easier for users to use. 

did not know about the new automated source - any details? and the automated source basically "union all" the existing table on this repo with distinct records? thats awesome

**Comment by GoodDataBadGuy** [comment]
those token are not added by the automated source yet it seems. perhaps we can merge this PR thru and I will keep an eye on how the automated source has develop?

I will create PR unless i don't see new token being added until certain time though

**Pull request merged**